pkg/cmd/util/clientcmd/factory.go

```diff
@@ -375,8 +375,14 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 			return kLogsForObjectFunc(object, options)
 		}
 	}
-	w.Printer = func(mapping *meta.RESTMapping, noHeaders, withNamespace, wide bool, showAll bool, showLabels, absoluteTimestamps bool, columnLabels []string) (kubectl.ResourcePrinter, error) {
-		return describe.NewHumanReadablePrinter(noHeaders, withNamespace, wide, showAll, showLabels, absoluteTimestamps, columnLabels), nil
+	w.Printer = func(mapping *meta.RESTMapping, options *kubectl.PrintOptions) (kubectl.ResourcePrinter, error) {
+		if mapping != nil {
```

Comment:
Maybe this needs a comment or something, it's not immediately clear what this block means.
Why do we set `options.KindName = mapping.Resource`?! And later maybe change it again.

Today we know this is about force printing resource names (or short names) when there are multiple objects types in the same output list... but what about us maintaining this code a week from now? :)

Hmm, I'm not even sure my sentence above is right, as this code doesn't seem to represent it.

What is this meant to do? :)